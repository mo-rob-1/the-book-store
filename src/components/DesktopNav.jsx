function DesktopNav() {
  return (
    <nav className="desktop-nav">
      <ul>
        <li>
          <a href="/">Home</a>
        </li>
        <li>
          <a href="/">Books</a>
        </li>
        <li>
          <a href="/">Magazines</a>
        </li>
        <li>
          <a href="/">E-Books</a>
        </li>
        <li>
          <a href="/">Account</a>
        </li>
      </ul>
    </nav>
  );
}

export default DesktopNav;
