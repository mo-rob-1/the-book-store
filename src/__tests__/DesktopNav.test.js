import { render, screen } from "@testing-library/react";
import DesktopNav from "../components/DesktopNav";

describe("DesktopNav", () => {
  it("renders the desktop nav component", () => {
    const { container } = render(<DesktopNav />);
    expect(container).toMatchSnapshot();
  });

  it("renders the navigation", () => {
    render(<DesktopNav />);
    const navElement = screen.getByTestId("navigation");
    expect(navElement).toBeTruthy();
  });

  it("renders the Home text link", () => {
    render(<DesktopNav />);
    expect(screen.getByText("Home")).toBeInTheDocument();
  });

  it("renders the Books text link", () => {
    render(<DesktopNav />);
    expect(screen.getByText("Books")).toBeInTheDocument();
  });

  it("renders the Magazines text link", () => {
    render(<DesktopNav />);
    expect(screen.getByText("Magazines")).toBeInTheDocument();
  });

  it("renders the E-Books text link", () => {
    render(<DesktopNav />);
    expect(screen.getByText("E-Books")).toBeInTheDocument();
  });

  it("renders the Account text link", () => {
    render(<DesktopNav />);
    expect(screen.getByText("Account")).toBeInTheDocument();
  });
});
