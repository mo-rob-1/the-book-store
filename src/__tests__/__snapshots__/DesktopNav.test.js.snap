// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNav renders the desktop nav component 1`] = `
<div>
  <nav
    class="desktop-nav"
    data-testid="navigation"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Books
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Magazines
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          E-Books
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Account
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
